plugins/openwhisk/src/main/java/com/navercorp/pinpoint/plugin/openwhisk/OpenwhiskPlugin.java

```diff
@@ -0,0 +1,115 @@
+
+import java.security.ProtectionDomain;
+import java.util.Optional;
+
+/**
+ * @author Seonghyun Oh
+ */
+public class OpenwhiskPlugin implements ProfilerPlugin, TransformTemplateAware {
+
+    private TransformTemplate transformTemplate;
+
+    @Override
+    public void setup(ProfilerPluginSetupContext context) {
+        final OpenwhiskConfig config = new OpenwhiskConfig(context.getConfig());
+        if (!config.isEnable()) {
+            return;
+        }
+
+        OpenwhiskDetector openwhiskDetector = new OpenwhiskDetector();
+        context.addApplicationTypeDetector(openwhiskDetector);
+
+        transformTemplate.transform("whisk.http.BasicHttpService$$anonfun$assignId$1$$anonfun$apply$13", new TransformCallback() {
+            @Override
+            public byte[] doInTransform(Instrumentor instrumentor, ClassLoader classLoader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws InstrumentException {
+                final InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className, classfileBuffer);
+                final InstrumentMethod method = target.getDeclaredMethod("apply", "akka.http.scaladsl.server.RequestContext");
+                method.addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdCreateInterceptor");
+
+                return target.toBytecode();
+            }
+        });
+
+        transformTemplate.transform("whisk.common.TransactionId$", new TransformCallback() {
+            @Override
+            public byte[] doInTransform(Instrumentor instrumentor, ClassLoader classLoader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws InstrumentException {
+                final InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className, classfileBuffer);
+
+                Optional<InstrumentMethod> method = target.getDeclaredMethods()
+                        .stream()
+                        .filter(instrumentMethod -> instrumentMethod.getDescriptor().getMethodName().equals("started$extension"))
+                        .findFirst();
+
+                if (method.isPresent()) {
+                    method.get().addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdStartedInterceptor");
+                }
+
+                Optional<InstrumentMethod> method2 = target.getDeclaredMethods()
+                        .stream()
+                        .filter(instrumentMethod -> instrumentMethod.getDescriptor().getMethodName().equals("finished$extension"))
+                        .findFirst();
+
+                if (method2.isPresent()) {
+                    method2.get().addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdFinishedInterceptor");
+                }
+
+                return target.toBytecode();
+            }
+        });
```

Comment:
It would be better to include an interceptor in the failure state. (for failed())